internal/model/route_test.go

```diff
@@ -0,0 +1,332 @@
+package model
+
+import (
+	"github.com/google/uuid"
+	"kube-proxless/internal/utils"
+	"testing"
+	"time"
+)
+
+func TestNewRoute(t *testing.T) {
+	route := &Route{
+		id:         uuid.New().String(),
+		service:    "helloworld",
+		port:       "8080",
+		deployment: "helloworld",
+		namespace:  "helloworld",
+		domains:    []string{"helloworld.io"},
+	}
+
+	routeWithDefaultPort := *route
+	routeWithDefaultPort.port = "80"
+
+	testCases := []struct {
+		id        string
+		svc       string
+		port      string
+		deploy    string
+		ns        string
+		domains   []string
+		want      *Route
+		errWanted bool
+	}{
+		{ // route ok
+			route.id,
+			route.service,
+			route.port,
+			route.deployment,
+			route.namespace,
+			route.domains,
+			route,
+			false,
+		},
+		{ // service missing
+			route.id,
+			"",
+			route.port,
+			route.deployment,
+			route.namespace,
+			route.domains,
+			route,
+			true,
+		},
+		{ // deployment name missing
+			route.id,
+			route.service,
+			route.port,
+			"",
+			route.namespace,
+			route.domains,
+			route,
+			true,
+		},
+		{ // namespace missing
+			route.id,
+			route.service,
+			route.port,
+			route.deployment,
+			"",
+			route.domains,
+			route,
+			true,
+		},
+		{ // domains nil
+			route.id,
+			route.service,
+			route.port,
+			route.deployment,
+			route.namespace,
+			nil,
+			route,
+			true,
+		},
+		{ // domains empty
+			route.id,
+			route.service,
+			route.port,
+			route.deployment,
+			route.namespace,
+			[]string{},
+			route,
+			true,
+		},
+		{ // port empty -> must default to "80"
+			route.id,
+			route.service,
+			"",
+			route.deployment,
+			route.namespace,
+			route.domains,
+			&routeWithDefaultPort,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		got, errGot := NewRoute(tc.id, tc.svc, tc.port, tc.deploy, tc.ns, tc.domains)
+
+		if tc.errWanted != (errGot != nil) {
+			t.Errorf("CreateRoute(tc %s) = %v, errWanted = %t", tc.id, errGot, tc.errWanted)
+		}
+
+		if errGot == nil && !compareRoute(got, tc.want) {
+			t.Errorf("CreateRoute(tc %s) - got and want does not match", tc.id)
+		}
+	}
+}
+
+func TestRoute_useDefaultPortIfEmpty(t *testing.T) {
+	testCases := []struct {
+		p    string
+		want string
+	}{
+		{"8080", "8080"},
+		{"", "80"},
+	}
+
+	for _, tc := range testCases {
+		got := useDefaultPortIfEmpty(tc.p)
+
+		if got != tc.want {
+			t.Errorf("useDefaultPortIfEmpty(%s) = %s, want %s", tc.p, got, tc.want)
+		}
+	}
+}
+
+// TODO lot of duplicate code becuz of getter and setter - see if we can factorize
+
+func TestRoute_SetLastUsed(t *testing.T) {
+	route := Route{}
+	now := time.Now()
+	route.SetLastUsed(now)
+
+	if route.lastUsed != now {
+		t.Errorf("SetLastUsed(%s) != want %s", route.lastUsed, now)
+	}
+}
+
+func TestRoute_GetDeployment(t *testing.T) {
```

Comment:
comment - wow getter and setter tests? pretty hard core :)

Reply:
ahah I was expecting ur comments :P
basically after working for a while on KintoHub I am starting to hate the `NoConstructor` approach. Cuz the object can be modified everywhere and u can miss initialization of fields. So I chose to make the model bullet proof so that if a package wants to use it, the route are correct.